common/scripted_buttons/imperia_frankfurt_parliament_buttons.txt

```diff
@@ -0,0 +1,1885 @@
+	visible = {
+		has_global_variable = frankfurt_parliament_current_vote
+		global_var:frankfurt_parliament_current_vote = 0
+	}
+	ai_chance = {
+		base = 100
+	}
+	possible = {
+		custom_tooltip = {
+			text = frankfurt_parliament_joining_parliament
+			global_var:frankfurt_parliament_current_vote = 0
+		}
+		hidden_trigger = {
+			NOT = {
+				has_variable = frankfurt_parliament_joined_parliament
+			}
+		}
+	}
+	selected = {
+		has_variable = frankfurt_parliament_joined_parliament
+	}
+	effect = {
+		custom_tooltip = {
+			text = frankfurt_joins_the_parliament_tt
+			set_variable = frankfurt_parliament_joined_parliament
+			remove_variable ?= left_frankfurt_parliament
+		}
+	}
+}
+
+je_frankfurt_parliament_choose_club_button = {
+	name = "je_frankfurt_parliament_choose_club_button"
+	desc = "je_frankfurt_parliament_choose_club_button_desc"
+	visible = {
+		is_player = yes
+		has_global_variable = frankfurt_parliament_current_vote
+		global_var:frankfurt_parliament_current_vote < 6
+	}
+	ai_chance = {
+		base = 0
+	}
+	possible = {
+		custom_tooltip = {
+			text = club_change_cooldown_tt
+			NOT = {
+				has_variable = club_change_cooldown
+			}
+		}
+	}
+	selected = {
+		has_variable = club_change_cooldown
+	}
+	effect = {
+		trigger_event = {
+			id = frankfurt_parliament.6
+			popup = yes
+		}
+	}
+}
+
+je_frankfurt_parliament_vote_republic_button = {
+	name = "je_frankfurt_parliament_vote_republic_button"
+	desc = "je_frankfurt_parliament_vote_republic_button_desc"
+	visible = {
+		has_global_variable = frankfurt_parliament_current_vote
+		global_var:frankfurt_parliament_current_vote = 1
+	}
+	ai_chance = {
+		base = 0
+		modifier = {
+			trigger = {
+				ruler = {
+					interest_group = {
+						OR = {
+							prefers_law = {
+								law = law_type:law_presidential_republic
+								comparison_law = law_type:law_monarchy
+							}
+							prefers_law = {
+								law = law_type:law_parliamentary_republic
+								comparison_law = law_type:law_monarchy
+							}
+						}
+					}
+				}
+			}
+			add = 75
+		}
+		modifier = {
+			trigger = {
+				any_country = {
+					has_journal_entry = je_imperia_frankfurt_parliament
+					relations:root >= relations_threshold:cordial
+					country_rank >= rank_value:major_power
+					has_variable = frankfurt_parliament_voted_republic
+				}
+			}
+			add = 20
+		}
+		modifier = {
+			trigger = {
+				any_country = {
+					has_journal_entry = je_imperia_frankfurt_parliament
+					relations:root >= relations_threshold:amicable
+					country_rank >= rank_value:major_power
+					has_variable = frankfurt_parliament_voted_republic
+				}
+			}
+			add = 50
+		}
+		modifier = {
+			trigger = {
+				any_country = {
+					has_journal_entry = je_imperia_frankfurt_parliament
+					relations:root >= relations_threshold:friendly
+					country_rank >= rank_value:major_power
+					has_variable = frankfurt_parliament_voted_republic
+				}
+			}
+			add = 80
+		}
```

Comment:
How do these even work? The conditions are cyclical... Is this only for players?

Reply:
The higher the relations with a major german power that voted for x option, the more likely they are to vote for said option.
If they dont have domestic support through the clubs however they wont vote for it at all. And once they DO have voting support they will swing their vote hard.
This is to prevent flip-flopping, although I did notice that a few votes were constantly swinging around after I added the "live" vote view.. So Im not sure whats up there.

Reply:
> The higher the relations with a major german power that voted for x option, the more likely they are to vote for said option. If they dont have domestic support through the clubs however they wont vote for it at all. And once they DO have voting support they will swing their vote hard. This is to prevent flip-flopping, although I did notice that a few votes were constantly swinging around after I added the "live" vote view.. So Im not sure whats up there.

So their ai chance depends on the other german majors, which depends on other german majors etc.

Reply:
Hmm, so perhaps these should not be relevant for major powers, only for lesser german princes?